File: intro/intro.hs
import System.IO

main :: IO ()
main = do
    hSetBuffering stdout NoBuffering -- DO NOT REMOVE
    
    loop

loop :: IO ()
loop = do
    input_line <- getLine
    let enemy1 = input_line :: String -- name of enemy 1
    input_line <- getLine
    let dist1 = read input_line :: Int -- distance to enemy 1
    input_line <- getLine
    let enemy2 = input_line :: String -- name of enemy 2
    input_line <- getLine
    let dist2 = read input_line :: Int -- distance to enemy 2
    
    -- Enter the code here

    if (dist1 < dist2) then
        putStrLn enemy1
    else
        putStrLn enemy2    
    loop

File: intro/intro.vb
Module Player

    Sub Main ()
        
        ' game loop
        While True
            Dim enemy1 as String
            enemy1 = Console.ReadLine() ' name of enemy 1

            Dim dist1 as Integer
            dist1 = Console.ReadLine() ' distance to enemy 1

            Dim enemy2 as String
            enemy2 = Console.ReadLine() ' name of enemy 2

            Dim dist2 as Integer
            dist2 = Console.ReadLine() ' distance to enemy 2

            ' Write an action using Console.WriteLine()

            ' Enter the code here

            If dist1 < dist2 Then
                Console.WriteLine(enemy1)
            Else
                Console.WriteLine(enemy2)
            End If
        End While
    End Sub
End Module
